### PR TITLE
translate configmap into chinese

### PR DESCRIPTION
**Pull request by fancc**
translate configmap into chinese

### CODE REVIEW

File: content/zh/docs/concepts/configuration/configmap.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+that lets you store configuration for other objects to use. Unlike most
+Kubernetes objects that have a `spec`, a ConfigMap has a `data` section to
+store items (keys) and their values.
+
+The name of a ConfigMap must be a valid
+[DNS subdomain name](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names).
+-->
+## ConfigMap 对象
+
+ConfigMap 是一个 API [对象](/docs/concepts/overview/working-with-objects/kubernetes-objects/)，让你可以存储其他对象所需要使用的配置。和其他 Kubernetes 对象都有一个 `spec` 不同的是，ConfigMap 使用 `data` 块来存储一个键和它的值。
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:62]:
```suggestion
ConfigMap 是一个 API [对象](/docs/concepts/overview/working-with-objects/kubernetes-objects/)，让你可以存储其他对象所需要使用的配置。和其他 Kubernetes 对象都有一个 `spec` 不同的是，ConfigMap 使用 `data` 块来存储元素（键名）和它们的值。
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+in that Pod based on the data in the ConfigMap. The Pod and the ConfigMap must be in
+the same {{< glossary_tooltip text="namespace" term_id="namespace" >}}.
+
+Here's an example ConfigMap that has some keys with single values,
+and other keys where the value looks like a fragment of a configuration
+format.
+-->
+## ConfigMaps 和 Pods
+
+您可以写一个引用 ConfigMap 的 Pod 的 `spec`，并根据 ConfigMap 中的数据在该 Pod 中配置容器。这个 Pod 和 ConfigMap 必须要在同一个 {{< glossary_tooltip text="namespace" term_id="namespace" >}} 中。
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:79]:
```suggestion
您可以写一个引用 ConfigMap 的 Pod 的 `spec`，并根据 ConfigMap 中的数据在该 Pod 中配置容器。这个 Pod 和 ConfigMap 必须要在同一个 {{< glossary_tooltip text="名字空间" term_id="namespace" >}} 中。
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+1. Add a file in read-only volume, for the application to read
+1. Write code to run inside the Pod that uses the Kubernetes API to read a ConfigMap
+
+These different methods lend themselves to different ways of modeling
+the data being consumed.
+For the first three methods, the
+{{< glossary_tooltip text="kubelet" term_id="kubelet" >}} uses the data from
+the Secret when it launches container(s) for a Pod.
+-->
+您可以使用四种方式来使用 ConfigMap 在 Pod 内配置一个容器：
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:117]:
```suggestion
您可以使用四种方式来使用 ConfigMap 配置 Pod 中的容器：
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+1. 容器入口点的命令行参数
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:119]:
```suggestion
1. 容器 entrypoint 的命令行参数
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+1. 容器的环境变量
+1. 在只读卷里面添加一个文件，让应用来读取
+1. 编写代码在能够使用 Kubernetes API 来读取 ConfigMap 的 Pod 中运行
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:122]:
```suggestion
1. 编写代码在 Pod 中运行，使用 Kubernetes API 来读取 ConfigMap
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+
+这些不同的方法适用于不同的方式来对数据进行建模。对前三个方法，{{< glossary_tooltip text="kubelet" term_id="kubelet" >}} 使用 Secret 中的数据在 Pod 中启动容器。
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:124]:
```suggestion
这些不同的方法适用于不同的数据使用方式。对前三个方法，{{< glossary_tooltip text="kubelet" term_id="kubelet" >}} 使用 Secret 中的数据在 Pod 中启动容器。
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+apiVersion: v1
+kind: Pod
+metadata:
+  name: configmap-demo-pod
+spec:
+  containers:
+    - name: demo
+      image: game.example/demo-game
+      env:
+        # Define the environment variable
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:149]: 这里的注释是可以翻译的

File: content/zh/docs/concepts/configuration/configmap.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+The most common way to use ConfigMaps is to configure settings for
+containers running in a Pod in the same namespace. You can also use a
+ConfigMap separately.
+
+For example, you
+might encounter {{< glossary_tooltip text="addons" term_id="addons" >}}
+or {{< glossary_tooltip text="operators" term_id="operator-pattern" >}} that
+adjust their behavior based on a ConfigMap.
+-->
+ConfigMaps 最常见的用法是为同一命名空间里的 Pod 中运行的容器配置设置。您也可以单独使用 ConfigMap。
```
Comment (tengqm) [inline content/zh/docs/concepts/configuration/configmap.md:211]:
```suggestion
ConfigMaps 最常见的用法是为同一命名空间里某 Pod 中运行的容器执行配置。您也可以单独使用 ConfigMap。
```

**Comment by tengqm** [comment]
nice.
/lgtm

File: content/zh/docs/concepts/configuration/configmap.md [hunk 9]
```diff
@@ -0,0 +1,229 @@
+---
+title: ConfigMaps
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:2]: nit (OK to merge without fixing)
```suggestion
title: ConfigMap
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+content_template: templates/concept
+weight: 20
+---
+
+{{% capture overview %}}
+
+{{< glossary_definition term_id="configmap" prepend="A ConfigMap is" length="all" >}}
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:9]: This text is not translated

File: content/zh/docs/concepts/configuration/configmap.md [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+
+这是一个 ConfigMap 的示例，它的一些键只有一个值，其他键的值看起来像是配置的片段格式。
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  Name: game-demo
+data:
+  # property-like keys; each key maps to a simple value
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:89]: optionally, translate this comment

File: content/zh/docs/concepts/configuration/configmap.md [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+  player_initial_lives: 3
+  ui_properties_file_name: "user-interface.properties"
+  #
+  # file-like keys
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:93]: optionally, translate this comment

File: content/zh/docs/concepts/configuration/configmap.md [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+The most common way to use ConfigMaps is to configure settings for
+containers running in a Pod in the same namespace. You can also use a
+ConfigMap separately.
+
+For example, you
+might encounter {{< glossary_tooltip text="addons" term_id="addons" >}}
+or {{< glossary_tooltip text="operators" term_id="operator-pattern" >}} that
+adjust their behavior based on a ConfigMap.
+-->
+ConfigMaps 最常见的用法是为同一命名空间里某 Pod 中运行的容器执行配置。您也可以单独使用 ConfigMap。
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:210]: nit:
```suggestion
ConfigMap 最常见的用法是为同一命名空间里某 Pod 中运行的容器执行配置。您也可以单独使用 ConfigMap。
```

File: content/zh/docs/concepts/configuration/configmap.md [hunk 14]
```diff
@@ -0,0 +1,229 @@
...
+{{% /capture %}}
+{{% capture whatsnext %}}
+
+<!--
+* Read about [Secrets](/docs/concepts/configuration/secret/).
+* Read [Configure a Pod to Use a ConfigMap](/docs/tasks/configure-pod-container/configure-pod-configmap/).
+* Read [The Twelve-Factor App](https://12factor.net/) to understand the motivation for
+  separating code from configuration.
+-->
+* 阅读 [Secrets](/docs/concepts/configuration/secret/)。
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:225]:
```suggestion
* 阅读 [Secret](/docs/concepts/configuration/secret/)。
```

**Comment by tengqm** [comment]
@fancc Would you address the feedbacks from Tim?

File: content/zh/docs/concepts/configuration/configmap.md [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+ConfigMap separately.
+
+For example, you
+might encounter {{< glossary_tooltip text="addons" term_id="addons" >}}
+or {{< glossary_tooltip text="operators" term_id="operator-pattern" >}} that
+adjust their behavior based on a ConfigMap.
+-->
+ConfigMaps 最常见的用法是为同一命名空间里某 Pod 中运行的容器执行配置。您也可以单独使用 ConfigMap。
+
+比如，您可能会遇到基于 ConfigMap 来调整其行为的 {{< glossary_tooltip text="addons" term_id="addons" >}} 或者 {{< glossary_tooltip text="operators" term_id="operator-pattern" >}}。
```
Comment (sftim) [inline content/zh/docs/concepts/configuration/configmap.md:212]: Possible also change addons to “addon”, and reword “operators”.

Comment (tengqm) [reply content/zh/docs/concepts/configuration/configmap.md:212]: addon can be translated to '插件', but there is no proper translation for 'operator' yet. Leaving it there should be okay.

Comment (sftim) [reply content/zh/docs/concepts/configuration/configmap.md:212]:
```suggestion
比如，您可能会遇到基于 ConfigMap 来调整其行为的 {{< glossary_tooltip text="插件" term_id="addons" >}} 或者 {{< glossary_tooltip text="operator" term_id="operator-pattern" >}}。
```
?

**Pull request merged**